annotations/src/main/java/com/amplifyframework/annotations/RestrictToAmplify.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.annotations
+
+/**
+ * API marked with this annotation is internal to Amplify, and it is not intended to be used outside.
+ * It could be modified or removed without any notice.
+ *
+ * We strongly recommend to not use such API.
+ */
+@Suppress("DEPRECATION")
+@RequiresOptIn(
+    level = RequiresOptIn.Level.ERROR,
+    message = "This API is internal to Amplify and should not be used. It could be removed or changed without notice.",
+)
+@Target(
+    AnnotationTarget.CLASS,
+    AnnotationTarget.TYPEALIAS,
+    AnnotationTarget.FUNCTION,
+    AnnotationTarget.PROPERTY,
+    AnnotationTarget.FIELD,
+    AnnotationTarget.CONSTRUCTOR,
+)
+public annotation class RestrictToAmplify
```

Comment:
Can we go with `InternalAmplifyApi` - that would be more consistent with other libraries.

Examples:
[InternalCoroutinesApi](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-internal-coroutines-api/)
[InternalComposeApi](https://developer.android.com/reference/kotlin/androidx/compose/runtime/InternalComposeApi)
[InternalSdkApi](https://github.com/awslabs/aws-sdk-kotlin/blob/177b444b561fa99a6eda0b641307e462cb192cb8/aws-runtime/aws-core/common/src/aws/sdk/kotlin/runtime/Annotations.kt) - aws kotlin sdk